sktime/transformations/series/theta.py

```diff
@@ -97,26 +97,26 @@ def _transform(self, X, y=None):
             pd.Series, with single Theta-line, if self.theta is float
             pd.DataFrame of shape: [len(X), len(self.theta)], if self.theta is tuple
         """
-        z = X
         theta = _check_theta(self.theta)
 
         forecaster = PolynomialTrendForecaster()
-        forecaster.fit(z)
-        fh = ForecastingHorizon(z.index, is_relative=False)
-        trend = forecaster.predict(fh)
+        forecaster.fit(y=X)
+        fh = ForecastingHorizon(X.index, is_relative=False)
+        trend = forecaster.predict(fh=fh)
 
-        theta_lines = np.zeros((z.shape[0], len(theta)))
+        theta_lines = np.zeros((X.shape[0], len(theta)))
         for i, theta in enumerate(theta):
-            theta_lines[:, i] = _theta_transform(z, trend, theta)
+            theta_lines[:, i] = _theta_transform(X, trend, theta)
         if isinstance(self.theta, (float, int)):
-            return pd.Series(theta_lines.flatten(), index=z.index)
+            return pd.Series(theta_lines.flatten(), index=X.index)
         else:
-            return pd.DataFrame(theta_lines, columns=self.theta, index=z.index)
+            return pd.DataFrame(theta_lines, columns=self.theta, index=X.index)
 
 
 def _theta_transform(Z, trend, theta):
     # obtain one Theta-line
     theta_line = Z * theta + (1 - theta) * trend
+    theta_line = theta_line.values.flatten()
```

Comment:
well, it happens when the input `X` to `_transform` is a `pd.DataFrame`, in that case `theta_line` will be `(N, 1)` which causes an exception right after the `_theta_transform` is called.

So, either `pd.DataFrame` has to be removed from the internal types (then the current code would run), or the internal code has to be made compatible.

I went with the option to changing internal functionality to the claimed guarantees, rather than restricting the claimed guarantees to the actual supported types.

Reply:
PS: I interpret this "approve" as restricted to the theta transformer, is that correct?
If so, I think your comments are more appropriate for the PR where the issue is fixed, that's this one:
https://github.com/alan-turing-institute/sktime/pull/1846

This PR relies on #1846, I've factored out the estimator bugfixes to there.

Reply:
merged-addressed in #1846, so resolved.